run.py

```diff
@@ -29,14 +29,14 @@ def main(gear_context):
 
     # Nipype interface output from dcm2niix can be a string or list (desired)
     if output is not None:
-        nifti_files = output.outputs.converted_files
+        output_image_files = output.outputs.converted_files
 
-        if isinstance(nifti_files, str):
-            nifti_files = [nifti_files]
+        if isinstance(output_image_files, str):
+            output_image_files = [output_image_files]
     else:
-        nifti_files = None
+        output_image_files = None
 
-    if not isinstance(nifti_files, list):
+    if not isinstance(output_image_files, list):
```

Comment:
Unless I am missing something, the logic above and the `list` or `str` output of `output.outputs.converted_files` guarantees that `output_image_files` will be either a `list` or a `None`.

If this is so, can this statement be reduced to:

`if not output_image_files:`

I am not familiar enough with the `output` to know what else there could be.

Reply:
I had to enforce some typing on the `output` object because it has a variable datatype return status. The `output` object is of type `nipype.interfaces.base.support.InterfaceResult` which can be a string or a list (as commented on line 30). From line 31 to line 37, if there is an output, then it is forced to a list datatype. This is because you can have one or many files in the output (think, diffusion has .bvecs and .bvals, but functional may only have one .nii) but we want to maintain the same datatype going forward. Hence, forcing the list datatype and defaulting to None, if not possible. From line 39 to line 51, it is really just ignoring the previous lines based on a config option. But I agree, there's a bit of a logic nightmare going on here. Some from the variable datatype return status of output object and some from config options. Something to add to the TODO list and think about how to make it cleaner on the next update round.